### PR TITLE
Artisan command

### PR DESCRIPTION
**Pull request by imanghafoori1**
Finally, I found the problem.

### CODE REVIEW

File: src/Laravel/TelegramServiceProvider.php [hunk 1]
```diff
@@ -48,6 +48,7 @@ public function register()
     {
         $this->registerManager($this->app);
         $this->registerBindings($this->app);
+        $this->commands('telegrambot.commands.webhook');
```
Comment (irazasyed) [inline src/Laravel/TelegramServiceProvider.php]: I think we should change this to `telegram.bot.commands.webhook`.

File: src/Laravel/TelegramServiceProvider.php [hunk 2]
```diff
@@ -80,6 +81,10 @@ protected function registerBindings(Application $app)
             return $manager->bot();
         });
 
+        $this->app->singleton('telegrambot.commands.webhook', function () {
```
Comment (irazasyed) [inline src/Laravel/TelegramServiceProvider.php]: Update this as well ofc.

File: src/Laravel/WebHookArtisanCommand.php [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+class WebHookArtisanCommand extends LaravelGeneratorCommand
+{
+    /**
+     * Create a new command instance.
+     *
+     */
+    public function __construct()
+    {
+        parent::__construct();
+    }
```
Comment (irazasyed) [inline src/Laravel/WebHookArtisanCommand.php]: Do we need the constructor for this command? I think you can omit it for now.

File: src/Laravel/TelegramServiceProvider.php (hunk 1 above)
Comment (imanghafoori1) [reply src/Laravel/TelegramServiceProvider.php]: This is a string. The dots as you know have no technical meaning.
It is a matter of human test to put dots.
I put them.

File: src/Laravel/WebHookArtisanCommand.php (hunk 3 above)
Comment (imanghafoori1) [reply src/Laravel/WebHookArtisanCommand.php]: For injecting other classes. Like Api class i think...

**Comment by imanghafoori1** [comment]
The last to commits are part of my tinkerings.
And have no effects

File: src/Laravel/WebHookArtisanCommand.php [hunk 4]
```diff
@@ -41,7 +44,9 @@ public function __construct()
     public function handle()
     {
         $botName = $this->argument('botName');
-        $this->token = config("telegram.bots.$botName.token");
+        $token = config("telegram.bots.$botName.token");
+        $this->telegram = new Api($token);
+
```
Comment (irazasyed) [inline src/Laravel/WebHookArtisanCommand.php]: Inject `BotsManager` and access the methods/config.

Example:
```php
use Telegram\Bot\BotsManager;

// ...
// ....

/** @var Telegram\Bot\Api */
protected $telegram;

/** @var array */
protected $config = [];

public function handle(BotsManager $telegram)
{
    $botName = $this->argument('botName');

    $this->telegram = $telegram->bot($botName);
    $this->config = $telegram->getBotConfig($botName);
}

private function setupWebhook()
{
    $webhookUrl = array_get($this->config, 'webhook_url');
    $certificatePath = array_get($this->config, 'certificate_path', false);

    $params = ['url' => $webhookUrl];
    
    if ($certificatePath) {
        $params['certificate'] = $certificatePath;
    }

    $response = $this->telegram->setWebhook($params);

    $this->info($response);
}
```

File: src/Laravel/config/telegram.php [hunk 5]
```diff
@@ -34,6 +34,8 @@
         'mybot' => [
             'username' => 'TelegramBot',
             'token'    => env('TELEGRAM_BOT_TOKEN', 'YOUR-BOT-TOKEN'),
+            'certificatePath'    => env('TELEGRAM_CERTIFICATE_PATH', 'YOUR-CERTIFICATE-PATH'),
+            'webhookUrl'    => env('TELEGRAM_WEBHOOK_URL', 'YOUR-BOT-WEBHOOK-URL'),
```
Comment (irazasyed) [inline src/Laravel/config/telegram.php]: Use snake case please! Ex: `webhook_url`

File: src/Laravel/WebHookArtisanCommand.php [hunk 6]
```diff
@@ -0,0 +1,93 @@
+<?php
+
+namespace Telegram\Bot\Laravel;
+
+use Illuminate\Console\GeneratorCommand as LaravelGeneratorCommand;
+use Symfony\Component\Console\Input\InputOption;
+use Telegram\Bot\Api;
+use Telegram\Bot\BotsManager;
+
+class WebHookArtisanCommand extends LaravelGeneratorCommand
```
Comment (irazasyed) [inline src/Laravel/WebHookArtisanCommand.php]: Please rename this to `WebhookCommand` (Notice **Webhook** Not **WebHook**). Obviously update everywhere.

Perhaps, we could also move this into its own namespace like `Telegram\Bot\Laravel\Artisan` as I think we can add some more useful commands in near future. Thanks!

**Comment by irazasyed** [comment]
Thanks 👍

**Pull request merged**